### PR TITLE
resending the pr - Ide 463

### PR DESCRIPTION
**Pull request by ligasm**

### CODE REVIEW

File: tools/portlet_spring_mvc_tmpl/docroot/WEB-INF/jsp/init.jspf [hunk 1]
```diff
@@ -0,0 +1,20 @@
+<%@page contentType="text/html" pageEncoding="UTF-8"%>
```
Comment (gamerson) [inline tools/portlet_spring_mvc_tmpl/docroot/WEB-INF/jsp/init.jspf]: Can we simplify the template to only include a single jsp file instead of a file and a fragment?  that will make it more consistent with other templates.

File: tools/portlet_spring_mvc_tmpl/docroot/WEB-INF/jsp/view.jsp [hunk 2]
```diff
@@ -0,0 +1,8 @@
+<%@page contentType="text/html" pageEncoding="UTF-8"%>
+
+<%@include file="../init.jspf"%>
+
+
```
Comment (gamerson) [inline tools/portlet_spring_mvc_tmpl/docroot/WEB-INF/jsp/view.jsp]: Lets make the default message to be the same as other default messages in other jsps, see the standard mvc for example.

File: tools/portlet_spring_mvc_tmpl/docroot/WEB-INF/spring-context/portlet-application-context.xml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+
+    <!--Enables @Autowired annotation-->
+    <context:annotation-config/>
+
+    <!-- Spring MVC VIEW Configuration -->
+    <bean id="viewResolver" class="org.springframework.web.servlet.view.InternalResourceViewResolver">
+        <property name="viewClass" value="org.springframework.web.servlet.view.JstlView"/>
+        <property name="contentType" value="text/html;charset=UTF-8" />
+        <property name="prefix" value="/WEB-INF/jsp/" />
```
Comment (gamerson) [inline tools/portlet_spring_mvc_tmpl/docroot/WEB-INF/spring-context/portlet-application-context.xml]: Do the jsps have to be in this folder?  I know that it may be best practice for spring-mvc projects, but just to make it more consistent with our other templates, could the jsps be moved to the docroot?

File: tools/portlet_spring_mvc_tmpl/docroot/WEB-INF/jsp/view.jsp (hunk 2 above)
Comment (gamerson) [inline tools/portlet_spring_mvc_tmpl/docroot/WEB-INF/jsp/view.jsp]: But it would be good to show the release info since the controller has prepared that for the view.

File: tools/portlet_spring_mvc_tmpl/ivy.xml [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+	xsi:noNamespaceSchemaLocation="http://ant.apache.org/ivy/schemas/ivy.xsd"
+>
+	<info module="@portlet.name@-portlet" organisation="com.liferay">
+		<extends extendType="configurations,description,info" location="${sdk.dir}/ivy.xml" module="com.liferay.sdk" organisation="com.liferay" revision="latest.integration" />
+	</info>
+
+	<dependencies defaultconf="default">
+		<dependency org="org.springframework" name="spring-webmvc" rev="3.0.7.RELEASE"/>
+		<dependency org="org.springframework" name="spring-webmvc-portlet" rev="3.0.7.RELEASE"/>
+		<dependency conf="test->default" name="arquillian-junit-container" org="org.jboss.arquillian.junit" rev="1.1.3.Final" />
```
Comment (gamerson) [inline tools/portlet_spring_mvc_tmpl/ivy.xml]: Can we simplify the dependencies here to not include the test dependencies?  I know that these are useful for when doing testing, but none of the other templates setup testing dependencies at the moment, so I would like for us to not include them at this time.

**Comment by gamerson** [comment]
Thanks @ligasm  this looks great.  I made a few comments, let me know what you think.  Please resend the PR as soon as you have some updates.  We will begin looking at integration on the LiferayIDE side as well.

**Comment by gamerson** [comment]
Also, Miroslav, how is your maven fu夫 these days?  Could you also help to
create a new maven archetype for spring-mvc?  We would need that one as
well.

G

On Tue, Jan 13, 2015 at 4:44 PM, Miroslav Ligas <notifications@github.com>
wrote:

> ------------------------------
> You can merge this Pull Request by running
>
>   git pull https://github.com/ligasm/liferay-plugins IDE-463
>
> Or view, comment on, or merge it at:
>
>   https://github.com/gamerson/liferay-plugins/pull/8
> Commit Summary
>
>    - IDE-463 - Added template for Spring MVC portlet
>    - IDE-463 Modified create script to facilitate Spring MVC
>
> File Changes
>
>    - *M* portlets/build.xml
>    <https://github.com/gamerson/liferay-plugins/pull/8/files#diff-0> (61)
>    - *M* portlets/create.bat
>    <https://github.com/gamerson/liferay-plugins/pull/8/files#diff-1> (4)
>    - *M* portlets/create.sh
>    <https://github.com/gamerson/liferay-plugins/pull/8/files#diff-2> (4)
>    - *A* tools/portlet_spring_mvc_tmpl/docroot/WEB-INF/jsp/init.jspf
>    <https://github.com/gamerson/liferay-plugins/pull/8/files#diff-3> (20)
>    - *A* tools/portlet_spring_mvc_tmpl/docroot/WEB-INF/jsp/view.jsp
>    <https://github.com/gamerson/liferay-plugins/pull/8/files#diff-4> (8)
>    - *A*
>    tools/portlet_spring_mvc_tmpl/docroot/WEB-INF/spring-context/portlet-application-context.xml
>    <https://github.com/gamerson/liferay-plugins/pull/8/files#diff-5> (25)
>    - *A*
>    tools/portlet_spring_mvc_tmpl/docroot/WEB-INF/spring-context/portlet-context.xml
>    <https://github.com/gamerson/liferay-plugins/pull/8/files#diff-6> (13)
>    - *A*
>    tools/portlet_spring_mvc_tmpl/docroot/WEB-INF/src/PortletViewController.java
>    <https://github.com/gamerson/liferay-plugins/pull/8/files#diff-7> (31)
>    - *A* tools/portlet_spring_mvc_tmpl/docroot/WEB-INF/web.xml
>    <https://github.com/gamerson/liferay-plugins/pull/8/files#diff-8> (26)
>    - *A* tools/portlet_spring_mvc_tmpl/ivy.xml
>    <https://github.com/gamerson/liferay-plugins/pull/8/files#diff-9> (20)
>
> Patch Links:
>
>    - https://github.com/gamerson/liferay-plugins/pull/8.patch
>    - https://github.com/gamerson/liferay-plugins/pull/8.diff
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/gamerson/liferay-plugins/pull/8>.
>



-- 
Greg Amerson
Liferay Developer Tools
Liferay, Inc. www.liferay.com

File: tools/portlet_spring_mvc_tmpl/docroot/WEB-INF/jsp/init.jspf (hunk 1 above)
Comment (ligasm) [inline tools/portlet_spring_mvc_tmpl/docroot/WEB-INF/jsp/init.jspf]: OK

File: tools/portlet_spring_mvc_tmpl/docroot/WEB-INF/jsp/view.jsp (hunk 2 above)
Comment (ligasm) [inline tools/portlet_spring_mvc_tmpl/docroot/WEB-INF/jsp/view.jsp]: OK

File: tools/portlet_spring_mvc_tmpl/docroot/WEB-INF/spring-context/portlet-application-context.xml (hunk 3 above)
Comment (ligasm) [inline tools/portlet_spring_mvc_tmpl/docroot/WEB-INF/spring-context/portlet-application-context.xml]: This is a tricky thing to do. Form security and clean code perspective the jsp files needs always be hidden, because they wont work when called directly. For our PortletMVC it make sans to make the jsp callable because there is no real controller that prepares data (it's the overall old strut arppoach) and the data are fetched in scriplets in jsps.
Opposing to that you should not have any data fetching in the view and every call to business layer should be done trough Controller class. So you should not be able to call the jsp because they will not work.
Shell I move it or can it stay?

Comment (gamerson) [inline tools/portlet_spring_mvc_tmpl/docroot/WEB-INF/spring-context/portlet-application-context.xml]: I'm fine with it staying... since we do have a real controller in this framework as you mentioned.

**Comment by gamerson** [comment]
Regarding maven, don't worry about that for now... that can be a followup discussion.  Get it into the SDK is more important for now.  Also... do you know fi there is a ticket for this over on  LPS?  Just wondering, seems like there used to be.

File: tools/portlet_spring_mvc_tmpl/ivy.xml (hunk 4 above)
Comment (ligasm) [inline tools/portlet_spring_mvc_tmpl/ivy.xml]: Contrary all other template do have the tests included in master. That's the reason I left them there.

Comment (ligasm) [inline tools/portlet_spring_mvc_tmpl/ivy.xml]: The other question with dependencies is if we shell depend on the Spring version included in Liferay portal witch is ancient and everybody will certainly change it or we include some new version trough ivy.

Comment (gamerson) [inline tools/portlet_spring_mvc_tmpl/ivy.xml]: Ok I stand corrected, they must have added them on master sometime back.  yeah lets leave them in here now.

**Comment by ligasm** [comment]
I can also prepare the maven archetype. I'm primarily a maven guy :) and I going to have a look at the tickets if there is something there.

**Comment by ligasm** [comment]
Let me know what you thing abut my argument again some of the changes you proposed.

**Comment by gamerson** [comment]
Hey @ligasm Can you rebase this make a few of those changes we agreed on and then resend?  the latest commits of SDK have moved the templates to to a new folder. 

Thanks.
